### ISSUE TITLE
syncthing user service doesn't inherit system options

### ISSUE THREAD

**Issue by kr-nn** [NixOS/nixpkgs#338480]
When I configure services.syncthing it works as intended but I would like to instead of using the system service, enable user services.
But if for ex. I set guiAddress = "0.0.0.0:8384"
the user service is still set to 127.0.0.1:8384

Steps:
1. add `services.syncthing = { enable = true; guiAddress = "0.0.0.0:8483" };`
2. cat /etc/systemd/user/syncthing.service
3. cat /etc/systemd/system/syncthing.service
4. notice the difference in options

The user services should still be configurable in services.syncthing.X

@Denperidge

**Comment by rnhmjoj** [comment]
In short, you can't: the NixOS module only configures the system-wide service; the user service unit is provided by the upstream project and installed as-is. The latter is meant to be configured imperatively.

If you want you can run the system-wide service under your user and still configure declaratively from NixOS. home-manager has a syncthing module if you want to configure both independently.

**Comment by eclairevoyant** [comment]
You could probably also inherit the config as needed from `config.systemd.services.syncthing.serviceConfig`, I haven't tested that out though (and of course you'll have to not inherit settings unsupported by user services).

**Comment by kr-nn** [comment]
I mean the user service is "System-Wide" its not ~/.config/systemd.
Its /etc/systemd

All users can use the /etc/systemd/user/syncthing.service

The nixos module does allow configuring these user services. Right now I'm taking advantage of systemd.user.units.syncthing.text as a workaround.

**Comment by eclairevoyant** [comment]
Yeah we probably will eventually want a more "portable" services framework but for now we're sticking to configuring system services primarily.

**Comment by eclairevoyant** [comment]
What I'm saying is, NixOS modules generally configure `systemd.services.<whatever>` not `systemd.user.services.<whatever>`. If we want to configure _both_, IMO it should be done the same way for all the modules (that it might apply to) rather than ad-hoc kludge added to individual services.

**Comment by lopter** [comment]
@kr-nn, if you want to provision multiple system instances of Syncthing for multiple users, then you can use containers for that (here is an [example] from my config files).

[example]: https://github.com/lopter/clan-destiny/blob/6e7d33644f0410183c9c1f44539520cf9646fc25/library/nix/nixosModules/syncthing.nix#L119-L213
